R/covariance_plot.R

```diff
@@ -77,11 +76,49 @@ h_vectorization <- function(vcov_matrix, string = NULL) {
   timepoints <- h_get_timepoint_vars(vcov_matrix, string)
   time_point_distribution <- timepoints$col_time - timepoints$row_time
   rank_row <- as.numeric(as.factor(timepoints$row_time))
-  rank_col <- as.numeric(as.factor(timepoints$row_time))
+  rank_col <- as.numeric(as.factor(timepoints$col_time))
   lag <- rank_col - rank_row
   if (any(is.na(lag)) | any(is.na(time_point_distribution))) {
     warning("Verify you have used the string argument correctly or
   that the columns and rows of your input matrix are named as expected")
   }
   data.frame(Vect = vect, time_point_distribution, lag, rank_row, rank_col)
 }
+#' Function to graph covariance
+#'
+#' @description Plot of covariance or correlation structures as a function of lag or time. The covariance structure
+#' needs to be vectorized and lag or time distances computed
+#' @param vcov_matrix (`matrix`)\cr name of the input symmetric matrix.
+#' @param x_var (`string`)\cr can be "lag" or "time_point_distribution" for lag and time distance
+#' respectively. The default value is "lag".
+#' @param legend_position (`string`)\cr denoting where the legend should be shown. Option
+#' are c("topright","topleft","bottomright","bottomleft"). Default #' value #' is "topright"
```

Comment:
```suggestion
#' @param legend_position (`string`)\cr denoting where the legend should be shown.
```
no need in general to repeat arguments in the documentation, as the user sees that in the help page already, same for defaults